www/class/centreonLDAP.class.php

```diff
@@ -966,23 +966,36 @@ public function isSyncNeededAtLogin(int $arId = null, int $contactId): bool
                 $stmtLdapBaseSync->execute();
                 $ldapBaseSync = $stmtLdapBaseSync->fetch();
 
+                // setting a default value, if the user account was imported and the user has never login
+                if (!$contactData['contact_ldap_last_sync']) {
```

Comment:
maybe check with empty or array key exists, as if `contact_ldap_last_sync` is equal to 0 you'll go into this condition and reassign it a value of 0.

Reply:
contact_ldap_last_sync is empty on import. It is filled with timestamp after the first login.
So we require a default value for the 978's line check

Reply:
after the import of the user's account the contact_ldap_last_sync == 0
(contact_ldap_last_sync is an int(11) column)

+------------+----------------+------------------------+
| contact_id | contact_name   | contact_ldap_last_sync |
+------------+----------------+------------------------+
|         36 | centreon-ldap  |                      0 |
|         35 | centreon-ldap2 |             1626446218 |
+------------+----------------+------------------------+

Reply:
So I'll remove this check